src/utils/account-with-lockup.js

```diff
@@ -81,7 +76,15 @@ export async function transferAllFromLockup(missingAmount) {
         receiver_id: this.wrappedAccount.accountId
     }, BASE_GAS.mul(new BN(2)))
 
-    await this.deleteLockupAccountIfPossible(lockupAccountId)
+    const lockedBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_locked_amount'))
+    if (lockedBalance.eq(new BN(0))) {
+        await this.wrappedAccount.functionCall(lockupAccountId, 'unselect_staking_pool', {}, BASE_GAS.mul(new BN(2)))
```

Comment:
@vgrichina is it necessary to check staking/pending balances with the staking pool before unselecting? I.e. is there any risk associated with unselecting the staking pool?

Reply:
Yes, unselecting active pool would fail. 

Generally it might be ok for the scope of this method (as it also means that we cannot really transfer all from lockup).

However we'd need to handle error upstream and direct user to unstake and withdraw from staking pool.

I think it also makes sense to try unselecting staking pool after we check for whether staking pool is selected (i.e. when we know there is some pool selected).